calicoctl/commands/node.go

```diff
@@ -25,30 +25,37 @@ import (
 // Node function is a switch to node related sub-commands
 func Node(args []string) error {
 	var err error
-	doc := `Usage: calicoctl node <command> [<args>...]
-
-    status          Shows the status of the node. 
+	doc := `Usage: 
+	calicoctl node status 
+	calicoctl node diags [--log-dir=<LOG_DIR>]
 
+Options:
+    --help                  Show this screen.
+    status                  Shows the status of the node.
+    diags                   Collects diagnostic information.
+    --log-dir=<LOG_DIR>     The directory for logs [default: /var/log/calico] 
+	
 Description:
   Node specific commands for calicoctl
   
-  See 'calicoctl node --help' to read about a specific subcommand.`
+  See 'calicoctl node --help' to read about a specific subcommand.
+  `
 
 	arguments, err := docopt.Parse(doc, args, true, "calicoctl", false, false)
 	if err != nil {
 		return err
 	}
 
-	// If `--help` or `-h` is passed, then arguments[] will be nil
-	if arguments["<command>"] != nil {
-		command := arguments["<command>"].(string)
-		nodeArgs := append([]string{command}, arguments["<args>"].([]string)...)
+	// If `--help` or `-h` is passed, then arguments map will be empty
+	if len(arguments) > 0 {
+		logDir := append([]string{"diags"}, arguments["--log-dir"].(string))
 
-		switch command {
-		case "status":
-			err = node.Status(nodeArgs)
-		default:
-			fmt.Printf("Invalid option: %s\n", command)
+		if arguments["status"].(bool) {
+			err = node.Status()
+		} else if arguments["diags"].(bool) {
+			err = node.Diags(logDir)
```

Comment:
we are not processing args anymore in node.go because status doesn't need any args and diags needs a --log-dir. Adding <args> and passing it along would be confusing I think